blockchain/blockchain.go

```diff
@@ -622,6 +651,21 @@ func stateUpdateByHash(txn db.Transaction, hash *felt.Felt) (*core.StateUpdate,
 	})
 }
 
+func l1MsgHashes(txn db.Transaction, l1HandlerMsgHash *common.Hash) (*felt.Felt, error) {
+	var raw []byte
+	err := txn.Get(db.L1HandlerTxnHash.Key(l1HandlerMsgHash.Bytes()), func(val []byte) error {
+		if len(val) == 0 {
+			return db.ErrKeyNotFound
+		}
+		raw = val
+		return nil
+	})
+	if err != nil {
+		return nil, err
+	}
+	return new(felt.Felt).SetBytes(raw), nil
```

Comment:
Are we sure it'd fit 251-bytes of Felt?

Reply:
Yup, the value is a felt (l1 handler transaction hash).